lambda_functions/review_service/notifications.py

```diff
@@ -110,10 +110,12 @@ def notify_users(is_test, session, item):
                 notifications_successful = False
 
         if submission.mail:
-            try:
-                notify_mail_user(submission.mail, item, rating, rating_text)
-            except Exception:
-                notifications_successful = False
+            if submission.status == 'confirmed':
+                try:
+                    notify_mail_user(submission.mail, item,
+                                     rating, rating_text)
+                except Exception:
+                    notifications_successful = False
```

Comment:
This doesn't affect your changes but is a general comment on the methods.

It is generally a bad idea to use Exceptions to control the flow of a program.
Here, the `notify_mail_user` method already wraps the entire code in a try/except block, logs the error which is a good practice, as you want to handle errors as close to their root as possible.
But then the except block raises a new exception. This is usually pretty expensive in terms of processing time and makes the code harder to read/understand.

I would suggest to change the `notify_mail_user` and `notify_telegram_user` method to return a boolean that indicates if the action was successful or not.

e.g.

`def notify_mail_user(params) -> boolean:
    try:
         sendMail
         return True
    except Exception:
         logError
         return False`

We can discuss that however at another point of time.